ts/parts-map/HeatmapSeries.ts

```diff
@@ -514,7 +606,47 @@ seriesType<Highcharts.HeatmapSeriesOptions>(
 
         /* eslint-enable valid-jsdoc */
 
-    }, colorPointMixin)
+    },
+    colorPointMixin,
+    {
+        setState: function (
+            this: Highcharts.HeatmapPoint,
+            state?: string
+        ): void {
+            var point = this,
+                series = this.series,
+                points = series.points,
+                inactiveException = (series.options.states as any)
```

Comment:
This `any` is avoidable by `series.options.states || {}`.

Reply:
Actually this will not work, you have to make a deep availability check for the option to make sure it is set.

Reply:
I had some problems here, could you clarify this a little to me?
`series.options.states.inactive` object **is always set**, because it's defined above:
https://github.com/highcharts/highcharts/pull/11548/files#diff-f8b433ee2021a9296168918bf2cf2288R298

In API it's optional, but in `series.options` it's always present. I think there's no need for checking like this:

```js
inactiveException = series.options.states &&
    series.options.states.inactive &&
    series.options.states.inactive.except;
```

Reply:
For the options we will have to find something that suits our and user needs. For now everything is optional, because every option can be set or not. As I understand you, a user is not able to make this option undefined. Is this correct?

Reply:
No, no, I mean user can set options as undefined, but for the most (all?) options in Series class we have (in short):

```js
series.options = merge(
  defaultOptions.plotOptions.type,
  defaultOptions.plotOptions.series,
  userOptions.plotOptions.type,
  userOptions.series[index]
);
```

This way, once something is set in default options for series (`H.seriesType('type', default.plotOptions.type, ...)`) it is available later, for example, in the case above we might have:

```js
merge(
  { },
  { states: { inactive: { except: 'column-row' } } },
  undefined,
  { color: 'red' } // from:  "series: [{ color: 'red' }]"
);
```

And still, the resulting `series.options` will have `states.inactive.except` available. Of course if a user sets for example: `states: function () {}` it will throw errors but this is a misconfigured chart (and users with TS shouldn't be able to build project).

Reply:
TypeScript forces us to handle misconfigurations. This can already happen internally when making use of prototype applies with fake instances. Please do the tripple check.

Reply:
@bre1470 can this comment thread be closed as resolved?

Reply:
Yes, it resolved.